### PR TITLE
Fix object type

### PR DESCRIPTION
**Pull request by wibimaster**
The object "types" does not know the "object" type ; test on a iPhone 4S with iOS 8.3, got a big fail with undefined object.

This fix was test on the same iPhone :)

Thanks !

### CODE REVIEW

**Comment by wibimaster** [comment]
Ping, is the author always active ?

**Comment by axemclion** [comment]
@wibimaster Not sure what you meant by "author always active"

**Comment by wibimaster** [comment]
@axemclion Thanks for your answer, just to know if I have a chance to see my request read or if I have to fork the project ;)

**Comment by axemclion** [comment]
@wibimaster I did see your pull request, but it is going to be hard for me to find time to review it. I am hoping that a few others who are using the project review the request, and if they are good, I can merge it in.

File: src/Key.js [hunk 1]
```diff
@@ -167,6 +167,16 @@
             }
         },
 
+        // Objects are encoded as JSON strings.
+        object: {
+            encode: function(key) {
+                return collations.indexOf("array") + "-" + JSON.stringify(key);
```
Comment (davaxi) [inline src/Key.js]: Why `collations.indexOf("array")` and not `collations.indexOf("object")` ?

**Comment by wibimaster** [comment]
@axemclion Okay, I understand :) It's just an "obstacle course" for us because after updated your repo, we have to ask for an update on https://github.com/ABB-Austin/cordova-plugin-indexeddb-async ;)

@davaxi Thanks, I fix that ASAP, just an oversight :)

**Comment by wibimaster** [comment]
Done, should close #231 too ;)

**Comment by brettz9** [comment]
As mentioned in #231, I've adapted this PR and incorporated it into master. Please note that if you are using your PR alone, there are some unintended consequences in its mistakenly allowing objects or null as keys (as values they are ok), but I've fixed these in master. Feel free to comment further, add a new bug, etc. if you find any new issues. With the W3C tests just added, we can hopefully improve quality even further.

**Comment by brettz9** [comment]
Sorry, I was premature in adding my code. Please pull again from `master` if you've already updated, as I've fixed things to either throw a proper error or silently ignore, as per spec behavior. Please see my comment at https://github.com/axemclion/IndexedDBShim/issues/215#issuecomment-227218203 .

**Pull request closed without merging**